lib/promiscuous/new_relic/instrumentation.rb

```diff
@@ -8,24 +7,19 @@
   end
 
   executes do
-    Promiscuous::Subscriber::UnitOfWork.class_eval do
-      include NewRelic::Agent::Instrumentation::ControllerInstrumentation
+    if defined?(Promiscuous::BlackHole)
+      Promiscuous::Subscriber::UnitOfWork.class_eval do
+        include PromiscuousNewRelicInstrumented
 
-      alias_method :execute_operation_without_rpm, :execute_operation
-      def execute_operation(operation)
-        # We are not using the subscriber class name, because of polymorphism
-        # We only want the parent class basically
-        perform_action_with_newrelic_trace(:name => namespace_for_rpm(operation), :class_name => 'Subscriber', :force => true,
-                                           :category => "OtherTransaction/Promiscuous") do
-
-          execute_operation_without_rpm(operation)
-        end
+        instrument :process
+        newrelic_namespace { "Normcore2/#{message.base_type}/#{message.operation}" }
```

Comment:
I don't like `Normcore2` here, that doesn't apply to others who are using the gem